### PR TITLE
[feat][feat]: add management fee

### PR DESCRIPTION
**Pull request by alpanaca**
## Description
Collecting management fee every time there is an interaction with delta neutral vault

## Why?
Mount to feed.

### CODE REVIEW

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 1]
```diff
@@ -221,6 +230,20 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
 
+  /// @notice minting shares as a form of management fee to teasury account
+  function _mintFee() internal {
+    _mint(config.getTreasuryAddr(), pendingManagementFee());
+    lastFeeCollected = block.timestamp;
+  }
+
+  /// @notice Return amount of share pending for minting as a form of management fee
+  function pendingManagementFee() public view returns (uint256) {
+    if (lastFeeCollected == 0) return 0;
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:241]: if we not initialize `lastFeeCollected`, is this code work on this line?
or contract will default it as 0

File: contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol [hunk 2]
```diff
@@ -293,10 +293,10 @@ contract DeltaNeutralPancakeWorker02 is OwnableUpgradeable, ReentrancyGuardUpgra
   /// @dev Return the amount of BaseToken to receive.
   /// @param id The position ID to perform health check. Note: This worker implementation ignore ID as the worker has only one position.
   function health(uint256 id) external view override returns (uint256) {
-    uint256 _totalBalanceInUSD = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
-    (uint256 _tokenPrice, uint256 _lastUpdate) = priceHelper.getTokenPrice(address(baseToken));
+    (uint256 _totalBalanceInUSD, uint256 _lpPriceLastUpdate) = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
+    (uint256 _tokenPrice, uint256 _tokenPricelastUpdate) = priceHelper.getTokenPrice(address(baseToken));
     // NOTE: last updated price should not be over 30 mins
-    if (block.timestamp - _lastUpdate > PRICE_AGE_SECOND) revert UnTrustedPrice();
+    if (block.timestamp - _lpPriceLastUpdate > PRICE_AGE_SECOND || block.timestamp - _tokenPricelastUpdate > PRICE_AGE_SECOND) revert UnTrustedPrice();
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol:299]: could u help change `PRICE_AGE_SECOND` to 1800 with comment, and remove constant `PRICE_AGE_SECOND` above

File: contracts/8.10/protocol/workers/DeltaNeutralMdexWorker02.sol [hunk 3]
```diff
@@ -292,10 +292,10 @@ contract DeltaNeutralMdexWorker02 is OwnableUpgradeable, ReentrancyGuardUpgradea
   /// @dev Return the amount of BaseToken to receive.
   /// @param id The position ID to perform health check. Note: This worker implementation ignore ID as the worker has only one position.
   function health(uint256 id) external view override returns (uint256) {
-    uint256 _totalBalanceInUSD = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
-    (uint256 _tokenPrice, uint256 _lastUpdate) = priceHelper.getTokenPrice(address(baseToken));
+    (uint256 _totalBalanceInUSD, uint256 _lpPriceLastUpdate) = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
+    (uint256 _tokenPrice, uint256 _tokenPricelastUpdate) = priceHelper.getTokenPrice(address(baseToken));
     // NOTE: last updated price should not be over 30 mins
-    if (block.timestamp - _lastUpdate > PRICE_AGE_SECOND) revert UnTrustedPrice();
+    if (block.timestamp - _lpPriceLastUpdate > PRICE_AGE_SECOND || block.timestamp - _tokenPricelastUpdate > PRICE_AGE_SECOND) revert UnTrustedPrice();
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/workers/DeltaNeutralMdexWorker02.sol:298]: could u change `PRICE_AGE_SECOND` to 1800 with comment, and remove constant `PRICE_AGE_SECOND` above

**Review by sora-alpaca — CHANGES_REQUESTED** [review]

File: contracts/8.10/protocol/DeltaNeutralVault.sol (hunk 1 above)
Comment (alpanaca) [reply contracts/8.10/protocol/DeltaNeutralVault.sol:241]: contract default as 0.

File: contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol (hunk 2 above)
Comment (alpanaca) [reply contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol:299]: I think it'll be in another PR that @jr-alpaca is working on.

File: contracts/8.10/protocol/workers/DeltaNeutralMdexWorker02.sol (hunk 3 above)
Comment (alpanaca) [reply contracts/8.10/protocol/workers/DeltaNeutralMdexWorker02.sol:298]: Same as above

File: contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol [hunk 4]
```diff
@@ -293,11 +292,11 @@ contract DeltaNeutralPancakeWorker02 is OwnableUpgradeable, ReentrancyGuardUpgra
...
   function health(uint256 id) external view override returns (uint256) {
-    uint256 _totalBalanceInUSD = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
-    (uint256 _tokenPrice, uint256 _lastUpdate) = priceHelper.getTokenPrice(address(baseToken));
+    (uint256 _totalBalanceInUSD, uint256 _lpPriceLastUpdate) = priceHelper.lpToDollar(totalLpBalance, address(lpToken));
+    (uint256 _tokenPrice, uint256 _tokenPricelastUpdate) = priceHelper.getTokenPrice(address(baseToken));
     // NOTE: last updated price should not be over 30 mins
-    if (block.timestamp - _lastUpdate > PRICE_AGE_SECOND) revert UnTrustedPrice();
-    // TODO: discuss round up or down
+    if (block.timestamp - _lpPriceLastUpdate > 1800 || block.timestamp - _tokenPricelastUpdate > 1800) revert UnTrustedPrice();
+    // TODO: discuss round up or dow
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/workers/DeltaNeutralPancakeWorker02.sol:299]:
```suggestion
    // TODO: discuss round up or down
```

**Review by sora-alpaca — CHANGES_REQUESTED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Review by sora-alpaca — APPROVED** [review]

**Pull request merged**